### PR TITLE
Cleanup Makefile

### PR DESCRIPTION
**Pull request by nojb**
Routine spring-cleaning. @mshinwell

### CODE REVIEW

File: .depend [hunk 1]
```diff
@@ -6073,6 +6073,3 @@ driver/compdynlink.cmo : \
     driver/compdynlink_common.cmi \
     bytecomp/cmo_format.cmi \
     driver/compdynlink.cmi
```
Comment (nojb) [inline .depend:6075]: These dependencies were repeated.

File: Makefile [hunk 2]
```diff
@@ -1212,47 +1212,9 @@ driver/compdynlink.mli: $(DYNLINK_DIR)/dynlink.mli \
...
-
-driver/compdynlink_types.cmo: driver/compdynlink_types.ml \
-    driver/compdynlink_types.cmi
-	$(CAMLC) $(COMPFLAGS) -c $<
-
-driver/compdynlink_types.cmx: driver/compdynlink_types.ml \
-    driver/compdynlink_types.cmi
-	$(CAMLOPT) $(COMPFLAGS) -c $<
-
 # See comment in otherlibs/dynlink/Makefile about these two rules.
```
Comment (nojb) [inline Makefile:1215]: These rules are subsumed by the general compilation rules and the computed dependencies in `.depend`.

File: Makefile [hunk 3]
```diff
@@ -1212,47 +1212,9 @@ driver/compdynlink.mli: $(DYNLINK_DIR)/dynlink.mli \
...
-
-driver/compdynlink_common.cmo: driver/compdynlink_common.ml \
-    driver/compdynlink_common.cmi \
-    driver/compdynlink_platform_intf.cmo
-	$(CAMLC) $(COMPFLAGS) -c $<
-
-driver/compdynlink_common.cmx: driver/compdynlink_common.ml \
-    driver/compdynlink_common.cmi \
-    driver/compdynlink_platform_intf.cmx
-	$(CAMLOPT) $(COMPFLAGS) -c $<
```
Comment (nojb) [inline Makefile:1218]: Same.

File: Makefile [hunk 4]
```diff
@@ -1265,6 +1227,7 @@ driver/compdynlink.cmx: driver/compdynlink.mlopt driver/compdynlink.cmi \
 beforedepend:: driver/compdynlink.mlbyte \
                driver/compdynlink.mlopt \
                driver/compdynlink_platform_intf.ml \
+               driver/compdynlink_platform_intf.mli \
```
Comment (nojb) [inline Makefile:1228]: To avoid the separate `ocamldep` call below.

**Comment by shindere** [comment]
Thanks a lot, Nicolas.

About your first commit, I am wondering: how aobut adding a warning to
ocamldep to let the user know it has been called on a non-existent file?
Not saying this should be implemented as part of this PR, just asking.

And, also about the first commit: did these files exist one day? What
happened that made them not exist any longer?

Very happy about the other commits!

File: .depend (hunk 1 above)
Comment (gasche) [reply .depend:6075]: Isn't this file auto-generated? Why is there a repetition here?

Comment (gasche) [reply .depend:6075]: Ah, I suppose the removal diff is not manual, but a result of the Makefile changes you did below.

Comment (nojb) [reply .depend:6075]: Exactly.

**Comment by nojb** [comment]
@shindere I think such a warning makes sense.

My uninformed guess is that those files existed in some intermediate state of #1063.

**Comment by dra27** [comment]
These changes are causing minor pain (or at least confusion) in #2041 - is there any objection to its being merged?

**Comment by nojb** [comment]
No! 😄 

The change is rather safe I think (it does not change the generated dependencies), and I have been working on some other PR on top of this change without issue.

**Comment by nojb** [comment]
Thanks!

**Pull request merged**